bluemira/base/parameter_frame/_frame.py

```diff
@@ -96,3 +96,33 @@ def from_json(cls, json_in: Union[str, json.SupportsRead]) -> NewParameterFrame:
                 return cls.from_dict(json.load(f))
         # load from a JSON string
         return cls.from_dict(json.loads(json_in))
+
+
+def parameter_setup(
+    params: Union[Dict, NewParameterFrame, None],
+    param_cls: Type[NewParameterFrame],
+) -> Union[Type[NewParameterFrame], None]:
```

Comment:
I think this should be `Union[NewParameterFrame, None]`.

I think we may need to do a bit of magic with the typing here - I suspect we'll need a `TypeVar`. Don't worry about that, though - I've got a branch that's attempting to fix all this typing stuff.